documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -1113,17 +1115,124 @@ public interface OrderMapper {
+
+MapStruct is able to handle `null` sources and `null` targets by means of the `<NULL>` keyword. 
+    
+[TIP]
+====
+Constants for `<ANY>`, `<ANY_UNMAPPED>` and `<NULL>` are available in the `MappingConstants` class.
+====
+        
+Finally `@InheritInverseConfiguration` and `@InheritConfiguration` can be used in combination with `@ValueMappings`.
+    
+.Enum mapping method, <NULL> and <ANY>
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+@Mapper
+public interface SpecialOrderMapper {
+
+    SpecialOrderMapper INSTANCE = Mappers.getMapper( SpecialOrderMapper.class );
+
+    @ValueMappings({
+        @ValueMapping( source = MappingConstants.NULL, target = "DEFAULT" ),
+        @ValueMapping( source = "STANDARD", target = MappingConstants.NULL ),
+        @ValueMapping( source = MappingConstants.ANY, target = "SPECIAL" )
     })
     ExternalOrderType orderTypeToExternalOrderType(OrderType orderType);
 }
 ----
 ====
 
-Note that an error will be raised by MapStruct in case a constant of the source enum type does not have a corresponding constant with the same name in the target type and also is not mapped to another constant via `@Mapping`. This ensures that all constants are mapped in a safe and predictable manner.
+.Enum mapping method result, <NULL> and <ANY>
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+// GENERATED CODE
+public class SpecialOrderMapperImpl implements SpecialOrderMapper {
+
+    @Override
+    public ExternalOrderType orderTypeToExternalOrderType(OrderType orderType) {
+        if ( orderType == null ) {
+            return ExternalOrderType.DEFAULT;
+        }
+
+        ExternalOrderType externalOrderType_;
+
+        switch ( orderType ) {
+            case STANDARD: externalOrderType_ = null;
+            break;
+            case RETAIL: externalOrderType_ = ExternalOrderType.RETAIL;
+            break;
+            case B2B: externalOrderType_ = ExternalOrderType.B2B;
+            break;
+            default: externalOrderType_ = ExternalOrderType.SPECIAL;
+        }
+
+        return externalOrderType_;
+    }
+}
+----
+====
+    
+*Note:* MapStruct would have refrained from mapping the `RETAIL` and `B2B` when `<ANY_UNMAPPED>` was used instead of `<ANY>`.  
+
+
+[WARNING]
+====
+The mapping of enum to enum via the `@Mapping` annotation is *DEPRECATED*. It will be removed from future versions of MapStruct.
```

Comment:
Can you add: "Please adapt existing enum mapping methods to make use of `@ValueMapping` instead".